File: bench/fibo/fibo.lua
-- $Id: fibo.lua,v 1.2 2005-03-19 00:32:56 igouy-guest Exp $
-- http://www.bagley.org/~doug/shootout/

function fib(n)
    if (n < 2) then return(n) end
    return( fib(n-2) + fib(n-1) )
end

N = tonumber((arg and arg[1])) or 1
io.write(fib(N), "\n")

File: bench/fibo/fibo.java
// $Id: fibo.java,v 1.2 2005-03-19 00:32:56 igouy-guest Exp $
// http://www.bagley.org/~doug/shootout/

public class fibo {
    public static void main(String args[]) {
	int N = Integer.parseInt(args[0]);
	System.out.println(fib(N));
    }
    public static int fib(int n) {
	if (n < 2) return(n);
	return( fib(n-2) + fib(n-1) );
    }
}
